### PR TITLE
Remove compiler conditions

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp.Drawing/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

Removes all outdated `#if...#else` conditions from the code.

### CODE REVIEW

File: src/ImageSharp.Drawing/Shapes/PolygonClipper/PolygonOffsetter.cs [hunk 1]
```diff
@@ -77,7 +77,7 @@ public void Execute(float delta, PathsF solution)
         this.ExecuteInternal(delta);
         if (this.groupList.Count == 0)
         {
-            goto Error;
+            return;
```
Comment (JimBobSquarePants) [inline src/ImageSharp.Drawing/Shapes/PolygonClipper/PolygonOffsetter.cs:80]: I spotted this when trying to run a benchmark. I'd deviated from clippers default behavior for some reason.

File: src/ImageSharp.Drawing/Utilities/SortUtility.KeyValueSort.cs [hunk 2]
```diff
@@ -1,6 +1,7 @@
 // Copyright (c) Six Labors.
 // Licensed under the Six Labors Split License.
 
+using System.Numerics;
```
Comment (JimBobSquarePants) [inline src/ImageSharp.Drawing/Utilities/SortUtility.KeyValueSort.cs:4]:
```
@antonfirsov I spotted this comment. Are we able to replace this code with built in APIs now?

// Adapted from:
// https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/ArraySortHelper.cs
// If targeting .NET 5, we can call span based sort, but probably not worth it only for that API.
```

**Review by dlemstra — DISMISSED** [review]

File: src/ImageSharp.Drawing/Utilities/SortUtility.KeyValueSort.cs (hunk 2 above)
Comment (antonfirsov) [reply src/ImageSharp.Drawing/Utilities/SortUtility.KeyValueSort.cs:4]: Yes, we can now delete `SortUtility` and use `span.Sort()` instead.

Comment (JimBobSquarePants) [reply src/ImageSharp.Drawing/Utilities/SortUtility.KeyValueSort.cs]: Awesome! I didn't even know about the additional sort APIs.

**Review by antonfirsov — APPROVED** [review]

**Pull request merged**